### PR TITLE
Rspec proposal

### PR DESCRIPTION
**Pull request by darthjee**
This is a proposal about using RSpec for unit testing

My idea, if you are up to it, would be to rewrite the tests in rspec format so that later I could split the gnuplot file to organize the code to facilitate contributions.

Here I am also adding simplecov to check spec code coverage

### CODE REVIEW

File: spec/integration/arrtest_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+describe "Array Plot Example" do
+  let(:file_path)    { 'spec/tmp/array_plot.eps' }
+  let(:fixture_path) { 'spec/fixtures/plots/array_plot.eps' }
+
```
Comment (darthjee) [inline spec/integration/arrtest_spec.rb]: the idea of this test is rely on a pregenerated file when testing the whole integration

**Comment by rdp** [comment]
sounds reasonable, I could merge in a few days when it has quieted a bit,
or would you like to be a maintainer?

**Comment by darthjee** [comment]
Well, since this is the first proposal, I would sugest not to merge yet, and I will write the specs for everything and then we do just one merge, WDYT?

About being maintainer, I could do that yes, I have deeply interest in this gem, but I have no hurry , I will only have time to actually use it next year :P

**Comment by rdp** [comment]
let me know when you feel ready with it :)

**Comment by darthjee** [comment]
Sure, still a lot to cover. one thing though, I think I found a bug

```ruby
plot = Gnuplot::Plot.new
plot.title "my title"
plot.title "My NEW Title"

plot['title'
```

this sounds like it should return `My NEW Title` while it is returning `my title` still

if that is the case, I know where the fix should go and can prepare a parallel PR for that

**Comment by rdp** [comment]
Sure.

**Comment by darthjee** [comment]
Right now, the specs are covering 84%

There is still a lot to cover (84% is not a real metric)

Since I will go on vacation, I would merge this one and finish it in January

Also, this can help with any refactoring comming

**Comment by rdp** [comment]
Awesome, you rock!

**Pull request merged**